### DISCUSSION TITLE
3.0.1 に関する不具合や質問はこちら

### DISCUSSION THREAD

**Discussion by dongaba** [dongaba/TVerRec#67] [Issues]
### 3.0.1 Fixed

- サイトマップダウンロード時に内部ファイル名が表示される不具合を修正

**Full Changelog**: https://github.com/dongaba/TVerRec/compare/v3.0.0...v3.0.1

<hr /><em>This discussion was created from the release <a href='https://github.com/dongaba/TVerRec/releases/tag/v3.0.1'>3.0.1</a>.</em>

**Comment by papagamo** [comment]
3.0.1をmacで利用しています。
コンソールの下側に

Web request status [Web request completed. (Number of bytes processed: 581)                                          ]  
Web request status [Web request completed. (Number of bytes processed: 1706)                                         ]

といったログが残る状態となっています。

**Reply by dongaba** [reply]
当方のMac環境ではPowershell7.4.1では再現できませんでしたので詳細わかりませんが、おそらくプログレスバーの名残だと思われます。
プログレスバーの表示のゴミが残ってしまうのはPowershellの不具合と思われ、現在有効な解決策を持っていません。
(過去にはCUIでもプログレスバーを表示する仕様としていましたが解決策が見つからないためプログレスバー表示をやめた経緯があります)

プログレスバーの表示は通常は行っておりませんので、なぜ表示されているのかについてはわかりません。(プログレスバーを使っているのはTVerRecの更新処理関連だけのはずです。)
もしPowershellのプロファイルを設定しているのであれば当方の意図と反してプログレスバーが出ている可能性は否定できません。

TVerRec更新時に出たプログレスバーが原因だと仮定すると、TVerRec更新時にはTVerRecを起動し直すと思うので、その際に`clear`を実行してターミナルを初期化するかターミナルを立ち上げ直せば問題は解決すると思います。

**Reply by papagamo** [reply]
ありがとうございます。
お教えいただいた方法で対応いたします。

**Comment by papagamo** [comment]
録画ファイルの移動ができていないケースがあります。
3.0.1をmacで利用しており、録画ファイルはNAS(NETGEAR　READYNAS)上に移動させています。
コンソール上に
```
Test-Path: "インストールディレクトリ"/TVerRec/src/move_video.ps1:112
Line |
 112 |          if (Test-Path $moveFromPath) {
     |              ~~~~~~~~~~~~~~~~~~~~~~~
     | Value cannot be null. (Parameter 'The provided Path argument was null or an empty collection.')

```
とメッセージが複数出ています。(移動が成功したものは通常と同様に番組名が表示されています。
`　"ダウンロード先ディレクトリ"/逆転裁判～その「真実」、異議あり！～ Season 2\*.mp4
`

OKだったものは「院内警察」「院内警察 解説放送版」「月とケーキ」「逆転裁判～その「真実」、異議あり！～ Season 2」など
NGだったものは「院内警察 ダイジェスト・予告」「不適切にもほどがある！ ダイジェスト・ナビ・予告」「アオハライド Season2」

手動で移動させれば、実害はありませんが何か考えられる要因などはありますでしょうか。

**Reply by dongaba** [reply]
Macをお使いとのことなので、Unicodeの正規化に起因する問題の可能性が高いです。
MacはUnicodeの正規化にNFDを使っており、端的に言うと濁点や半濁点を含む場合、2文字分のデータを使っています。
Windows/LinuxではNFCを使っているので、濁点や半濁点を含むファイル名、ディレクトリ名の場合にエラーになっていると思われます。
ダウンロード対象外リストとの照合時にはNFDとNFCの差を吸収する機能をv3.0.0から追加したのですが、フォルダ移動時にも同様の考慮が必要なのかもしれません。
ひとまずIssueとして管理させていただきます。

**Reply by dongaba** [reply]
v3.0.2で修正したつもりです。
